### ISSUE TITLE
Plotting error (Dimensions {'time'} do not exist)

### ISSUE THREAD

**Issue by thomaszwagerman** [davidwyld/amundsen-sea-low-index#23]
When running `a.plot_region_all()`:

```
ValueError: Dimensions {'time'} do not exist. Expected one or more of ('valid_time', 'latitude', 'longitude')
```

Error is raised when attempting to plot `a.masked_msl_data`.  This has the dimensions `'valid_time'`, `'latitude'`, `'longitude'`.

It goes wrong in `plot_lows()`:

```python
    for i in range(da.shape[0]):
        da_2D = da.isel(valid_time=i)
        # when indexing valid_time gets dropped.

[...]
      
       # But is referred to again here - not sure what this does!
       # Should da_2D be replaced with da and [0:7] with i?
       ax.set_title(str(da_2D.valid_time.values)[0:7])

        ## mark ASL
        df2 = df[df["time"] == str(da_2D.valid_time.values)[0:10]]
        if len(df2) > 0:
            ax.plot(df2["lon"], df2["lat"], "mx", transform=ccrs.PlateCarree())

        if regionbox:
            draw_regional_box(regionbox)

    return ax
```

I believe time should be changed to `valid_time`. We changed this variable name in relation to `expver` fixes, I believe.

**Comment by thomaszwagerman** [comment]
I think the origin of this is datetime information (%Y-%m-%d) information getting dropped from `a.masked_msl_data`. 

Maybe unintentional consequence of?

```python
        if hasattr(self.raw_msl_data, "expver") and not include_era5t:
            months = []
            for month in self.raw_msl_data:
                if month.expver.values == "0001":
                    months.append(month)
            self.raw_msl_data = xr.concat(months, dim="valid_time")
```

and then renaming valid_time to time in:

```python

        if "season" in self.sliced_msl.dims:
            ntime = 4
            slice_by = "season"
        if "valid_time" in self.sliced_msl.dims:
            self.sliced_msl = self.sliced_msl.rename({'valid_time': 'time'})
        if "time" in self.sliced_msl.dims:
            ntime = self.sliced_msl.time.shape[0]
            slice_by = "time"
```

**Comment by thomaszwagerman** [comment]
The above is not the origin of the datetime information being dropped.

Date gets dropped during:

```python
        self.masked_msl_data = self.raw_msl_data.where(
            self.land_sea_mask < MASK_THRESHOLD
        )
```

Not sure why, perhaps because `self.land_sea_mask` has no date attached, should this filtering be specified along specific variable?

**Comment by thomaszwagerman** [comment]
This prevents date from dropping:

```python
        # Using the .values, in order to prevent unintentional dropping of other attrs
        self.masked_msl_data = self.raw_msl_data.where(
            self.land_sea_mask.values < MASK_THRESHOLD
        )
```

But date remains an array of integers (rather than datetime), which remains a challenge for the date slicing in the plot_region_year() function

Something like:

```python
test_date_drop = self.raw_msl_data.where(
    self.land_sea_mask.values > 0.5
)

year_list = pd.date_range(str(year + '-01-01'), str(year + '-12-01'), 
              freq='MS').strftime("%Y%m%d").tolist()

da = test_date_drop.where(
    test_date_drop.date.values.any() in year_list
)
```

**Comment by davidwyld** [comment]
@thomaszwagerman Is this resolved by #24 ?

**Comment by thomaszwagerman** [comment]
Indeed :sunglasses:
